core/androidbp_backend.go

```diff
@@ -89,10 +92,33 @@ func androidBpSingletonFactory() blueprint.Singleton {
 
 func (s *androidBpSingleton) GenerateBuildActions(ctx blueprint.SingletonContext) {
 	sb := &strings.Builder{}
+
+	// read definitions of plugin packages
+	content, err := ioutil.ReadFile(filepath.Join(getBobDir(), "plugins/Android.bp.in"))
+	if err != nil {
+		utils.Exit(1, err.Error())
+	}
+
+	// use source dir to get project-unique identifier,
+	// generate 10 chars in hex based on its hash
+	h := sha1.New()
+	h.Write([]byte(getSourceDir()))
```

Comment:
OK - this works, because `srcdir` returns the path relative to the working directory, which on `Android.bp` is the root of  the Android tree. So e.g. we'd be hashing something like `external/bob-build/tests` here, which is enough to identify the project :)